anndata/_core/sparse_dataset.py

```diff
@@ -137,7 +138,7 @@ def _offsets(
     def _get_contiguous_compressed_slice(
         self, s: slice
     ) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
-        new_indptr = self.indptr[s.start : s.stop + 1]
+        new_indptr = self.indptr[s.start : s.stop + 1].copy()
```

Comment:
This is a bit of a downside @ivirshup but is probably better than re-reading into memory every time.

Reply:
This was not a "bug" before because `indptr` was always re-read in.  Now the issue has come to light.